OSDOCS-8253: Improved LokiStack integration

skrthomas opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->
For peer review/merge review:
Merge to only the `no-1.5` branch - no cherrypicks are required.
This PR is part of an experiment for simplifying merges for asynchronous content, and I will open one PR against main to incorporate all of the Network Observability 1.5 content just before its GA.

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->
4.11+
Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->
https://issues.redhat.com/browse/OSDOCS-8253
Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->
### API Version updates:
- [SRI-OV](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/configuring-operator#network-observability-SR-IOV-config_network_observability)
- [Disabling health alerts](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/network-observability-operator-monitoring#network-observability-disable-alerts_network_observability)
- [DNS Tracking](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/observing-network-traffic#network-observability-dns-tracking_nw-observe-network-traffic)
- [Enriched flows](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/configuring-operator#network-observability-enriched-flows_network_observability)
- [View the FlowCollector resource](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/configuring-operator#network-observability-flowcollector-view_network_observability)
- [FlowCollector Kafka Configuration](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/configuring-operator#network-observability-flowcollector-kafka-config_network_observability)
- [Working with packet drops](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/observing-network-traffic#network-observability-packet-drops_nw-observe-network-traffic)
- [Working with conversation tracking](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/observing-network-traffic#network-observability-working-with-conversations_nw-observe-network-traffic)

### Loki Installation step simplification
- [Installing the Network Observability Operator](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/installing-operators#network-observability-operator-installation_network_observability)
- [View the FlowCollector - callout #5 updated for LokiStack mode](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/configuring-operator#network-observability-flowcollector-view_network_observability)
- [Removed multi-tenancy auth info and updated prerequisites for multi-tenancy](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/installing-operators#network-observability-multi-tenancynetwork_observability)

### [IncludeList](https://issues.redhat.com/browse/NETOBSERV-1284)
[Network Observability metrics dashboards add includeList](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/network-observability-overview#network-observability-dashboards)

### Unrelated updates
[Packet Drop and DNS Overview nesting adjustment](https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/observing-network-traffic#network-observability-pktdrop-overview_nw-observe-network-traffic)

QE review:
- [X] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

skrthomas commented on modules/network-observability-operator-install.adoc:
```diff
@@ -34,17 +34,13 @@ This documentation assumes that your `LokiStack` instance name is `loki`. Using
-... *spec.loki.url*: Since authentication is specified separately, this URL needs to be updated to `https://loki-gateway-http.netobserv.svc:8080/api/logs/v1/network`. The first part of the URL, "loki", must match the name of your `LokiStack`.
-... *spec.loki.authToken*: Select the `FORWARD` value.
-... *spec.loki.statusUrl*: Set this to `https://loki-query-frontend-http.netobserv.svc:3100/`. The first part of the URL, "loki", must match the name of your `LokiStack`.
+... *spec.loki.mode*: Set this to `LokiStack` mode.
```
If I understood correctly from the Slack discussion, we're only documenting `LokiStack` mode, right?

skrthomas replied:
I have in my notes that `Manual` mode is just the old steps encapsulated in this spec:
 ```
 loki:
    mode: MANUAL
    manual:
      ingesterUrl: 'https://loki-gateway-http.netobserv.svc:8080/api/logs/v1/network/'
      statusUrl: 'https://loki-query-frontend-http.netobserv.svc:3100/'
      authToken: FORWARD
      tls:
        enable: true
        caCert:
          type: configmap
          name: loki-gateway-ca-bundle
          certFile: service-ca.crt
        insecureSkipVerify: false
      statusTls:
        enable: true
        caCert:
          certFile: service-ca.crt
          name: loki-ca-bundle
          type: configmap
        insecureSkipVerify: false
        userCert:
          certFile: tls.crt
          certKey: tls.key
          name: loki-query-frontend-http
          type: secret
 
```
So I agree, we don't need to go into these details I don't think. Unless we find more customers wanting complex Loki installation steps.

skrthomas replied:
Does this need documentation? Jw how customers would use these microservices? https://github.com/netobserv/network-observability-operator/blob/main/docs/FlowCollector.md#flowcollectorspeclokimicroservices

jpinsonneau replied:
Correct @skrthomas, we don't need to mention other modes here but it could be more detailed in `Troubleshooting Network Observability` for example.
@jotak would that work with your initial suggestion ?

As you pointed @skrthomas, everything is already described in [FlowCollector.spec.loki
](https://github.com/netobserv/network-observability-operator/blob/main/docs/FlowCollector.md#flowcollectorspecloki-2)

Example for `microservices`;
- set `mode` value to `Microservices`
- fill [FlowCollector.spec.loki.microservices](https://github.com/netobserv/network-observability-operator/blob/main/docs/FlowCollector.md#flowcollectorspeclokimicroservices) 
  - urls
  - tenant & tls (optionnal)

jotak replied:
Yes I agree; so the LokiStack mode should be documented here; I would say for the other modes, including Manual, the CRD doc should be sufficient. Maybe just mention here that other modes are available if users don't use the loki operator or if they want more tweakable install.

skrthomas commented:
@nathan-weinberg @jpinsonneau @acmenezes Here's my draft PR for Improved LokiStack configuration. Please let me know your feedback. I think I've got all the doc impacts from this jira, but let me know if I'm missing anything.

skrthomas commented on network_observability/network-observability-overview.adoc:
```diff
@@ -37,7 +37,7 @@ In *Observe* -> *Dashboards*, the *Netobserv* dashboard provides a quick overvie
 
 *Infrastructure* and *Application* metrics are shown in a split-view for namespace and workloads.
-You can configure the `FlowCollector` `spec.processor.metrics` to add or remove metrics by changing the `ignoreTags` list. For more information about available tags, see the xref:../network_observability/flowcollector-api.adoc#network-observability-flowcollector-api-specifications_network_observability[Flow Collector API Reference]
+You can configure the `FlowCollector` custom resource `spec.processor.metrics.includeList` to add or remove metrics. For more information about available tags, see the xref:../network_observability/flowcollector-api.adoc#network-observability-flowcollector-api-specifications_network_observability[Flow Collector API Reference]
```
Will the available list [items](https://github.com/netobserv/network-observability-operator/blob/main/docs/Metrics.md) be shown in the API reference? 🤔 Wondering if we should dedicate a section to documenting these list items

nathan-weinberg replied:
I would like to see that in the downstream docs personally! I think we should make it as easy as possible for customers to configure this metric since it's probably one of the most important ones in terms of what traffic is getting observed

skrthomas replied:
Cool, I was thinking the same. It could be helpful for them to have outside the API reference! Will doc it!

skrthomas commented:
I think the only thing I'm missing here WRT the [Easier Configuration ](https://issues.redhat.com/browse/NETOBSERV-763) is the [Move CRD fields to debug](https://issues.redhat.com/browse/NETOBSERV-1092), which is still in code review. Depending on troubleshooting needs for this, I may handle in a separate PR.

jpinsonneau commented on modules/network-observability-operator-install.adoc:
```diff
@@ -34,17 +34,13 @@ This documentation assumes that your `LokiStack` instance name is `loki`. Using
-... *spec.loki.authToken*: Select the `FORWARD` value.
-... *spec.loki.statusUrl*: Set this to `https://loki-query-frontend-http.netobserv.svc:3100/`. The first part of the URL, "loki", must match the name of your `LokiStack`.
+... *spec.loki.mode*: Set this to `LokiStack` mode.
+... *spec.loki.lokistack.name*: Set this to `LokiStack`.
```
Here we are pointing the `LokiStack` instance [previously created from yaml](https://docs.openshift.com/container-platform/4.14/network_observability/installing-operators.html#network-observability-lokistack-create_network_observability):
```
  apiVersion: loki.grafana.com/v1
  kind: LokiStack
  metadata:
    name: loki
    namespace: netobserv 
...
```

The name should be `loki` as:
```suggestion
... *spec.loki.lokistack.name*: Set this to `loki`.
```

Since the namespace is `netobserv` (the same as our operator), we don't need to mention `spec.loki.lokistack.namespace`. However, we should add a note that this can be updated as you did in LokiStack creation:
```
The installation examples in this documentation use the same namespace, netobserv, across all components. You can optionally use a different namespace.
```

nathan-weinberg replied:
@skrthomas if it is helpful, here are some examples of the `v1beta2` CRD
- official example in upstream operator repo: https://github.com/netobserv/network-observability-operator/blob/main/config/samples/flows_v1beta2_flowcollector.yaml
- QE CRD used in Perf pipeline: https://github.com/openshift-qe/ocp-qe-perfscale-ci/blob/netobserv-perf-tests/scripts/netobserv/flows_v1beta2_flowcollector.yaml

skrthomas replied:
@nathan-weinberg I was wondering where I got `LokiStack` from for thhe `spec.loki.lokistack.name` value, and I think I was looking at the 2nd CRD you linked; I think I found it in Jira. Anyways, the `spec.loki.lokistack.name` is `lokistack`...so is this wrong? Or do you guys just use `lokistack`
![image](https://github.com/openshift/openshift-docs/assets/77501157/8c25a056-82f8-4e73-bde4-cfd5194b3961)

nathan-weinberg replied:
we just used lokistack :sweat_smile: but it could be anything really

skrthomas replied:
Got it! 😆

jpinsonneau replied:
QE guys doesn't follow the docs :smiling_imp:

Amoghrd replied:
yeah we have been using name `lokistack` since the start but the docs and upstream mention `loki`. we dont follow the docs🫣

skrthomas replied:
I won't hold your feet to the fire 🤣  I think you can use what you want. Just for my personal information, I should remember this little naming thing when I'm looking at QE CRDs for doc updates

jotak commented on modules/network-observability-operator-install.adoc:
```diff
@@ -22,7 +22,7 @@ The actual memory consumption of the Operator depends on your cluster size and t
 [NOTE]
 ====
-This documentation assumes that your `LokiStack` instance name is `loki`. Using a different name requires additional configuration.
+This documentation assumes that your `LokiStack` instance name is `loki`. Using a different name requires additional configuration. Additionally, this installation example uses the `netobserv` namespace, which is used across all components. You can optionally use a different namespace.
```
I think we can remove this part "This documentation assumes that your `LokiStack` instance name is `loki`. Using a different name requires additional configuration."
Because now if they use a different LokiStack name, they just need to adapt the LokiStack reference and that's it, there shouldn't be other assumptions related to that name in the doc I think

jotak commented on modules/network-observability-operator-install.adoc:
```diff
@@ -34,17 +34,13 @@ This documentation assumes that your `LokiStack` instance name is `loki`. Using
-... *spec.loki.authToken*: Select the `FORWARD` value.
-... *spec.loki.statusUrl*: Set this to `https://loki-query-frontend-http.netobserv.svc:3100/`. The first part of the URL, "loki", must match the name of your `LokiStack`.
+... *spec.loki.mode*: Set this to `LokiStack` mode.
+... *spec.loki.lokistack.name*: Set this to `loki`.
```
I would say something like:

> set it to the name of your LokiStack resource (we used `loki` in the examples above).

So user understand we can accept any value

jotak commented on modules/network-observability-operator-install.adoc:
```diff
@@ -34,17 +34,13 @@ This documentation assumes that your `LokiStack` instance name is `loki`. Using
-... *spec.loki.statusUrl*: Set this to `https://loki-query-frontend-http.netobserv.svc:3100/`. The first part of the URL, "loki", must match the name of your `LokiStack`.
+... *spec.loki.mode*: Set this to `LokiStack` mode.
+... *spec.loki.lokistack.name*: Set this to `loki`.
 ... *spec.loki.tls.enable*: Select the checkbox to enable TLS.
 ... *spec.loki.statusTls*: The `enable` value is false by default.
```
Those two should also be removed

skrthomas replied:
Ah ok, I was following QE CRD https://github.com/openshift/openshift-docs/pull/67836#discussion_r1401262883, but I think some things are there that don't align quite with what we want to recommend in the doc. I will remove this. I was initially thinking this needed to be removed but then I saw it was still in the QE CRD

jotak commented on modules/network-observability-flowcollector-view.adoc:
```diff
@@ -17,7 +17,7 @@ The following example shows a sample `FlowCollector` resource for {product-title
 [source, yaml]
 ----
-apiVersion: flows.netobserv.io/v1beta1
+apiVersion: flows.netobserv.io/v1beta2
```
the `loki` section in the YAML should be amended to show the lokiStack mode

jotak commented:
Thanks @skrthomas !
I see more changes to do related to the Loki integration:

- in https://docs.openshift.com/container-platform/4.14/network_observability/installing-operators.html#network-observability-auth-mutli-tenancy_network_observability we can remove the whole section that mentions installing roles / role bindings: this is now done automatically
- in https://docs.openshift.com/container-platform/4.14/network_observability/installing-operators.html#network-observability-multi-tenancynetwork_observability , there is a prereq saying "The FlowCollector spec.loki.authToken configuration must be set to FORWARD.". Also the other prereq saying Loki operator 5.7 is needed should be amended to say "at least 5.7".
- At the bottom of https://docs.openshift.com/container-platform/4.14/network_observability/installing-operators.html#network-observability-operator-installation_network_observability , we should remove the notice saying to not use querierUrl.


Also we should briefly explain what happens under the hood when the LokiStack mode is used, in case people want to use manual mode they would better understand what they need to do. Something like:

> Setting the Loki mode to `LokiStack` will preconfigure various settings related to using the Loki Operator:
> - All Loki URL are set accordingly (Querier URL, Ingester URL and Status URL) along with their TLS configuration
> - Auth token is set to forward user token
> - Cluster roles and a cluster role binding are created for reading and writing logs to Loki

skrthomas commented:
@jotak @nathan-weinberg @jpinsonneau thank you for your reviews! Really good feedback and thanks for the help. I've addressed all of your points and updated the preview links to show the additional changes.

jotak reviewed (approved):
looks great! thanks

nathan-weinberg commented on modules/network-observability-flowcollector-view.adoc:
```diff
@@ -88,5 +79,5 @@ spec:
 <3> The optional specifications `spec.processor.logTypes`, `spec.processor.conversationHeartbeatInterval`, and `spec.processor.conversationEndTimeout` can be set to enable conversation tracking. When enabled, conversation events are queryable in the web console. The values for `spec.processor.logTypes` are as follows: `FLOWS` `CONVERSATIONS`, `ENDED_CONVERSATIONS`, or `ALL`. Storage requirements  ...
 <4> The Loki specification, `spec.loki`, specifies the Loki client. The default values match the Loki install paths mentioned in the Installing the Loki Operator section. If you used another installation method for Loki, specify the appropriate client information for your install.
-<5> The original certificates are copied to the Network Observability instance namespace and watched for updates. When not provided, the namespace defaults to be the same as "spec.namespace". If you chose to install Loki in a different namespace, you must specify it in the `spec.loki.tls.caCert.namespace` field.   Similarly, the `spec.exporters.kafka.tls.caCert.namespace` field is available for K ...
+<5> The `LokiStack` mode automatically sets a few configurations: `querierUrl`, `ingesterUrl` and `statusUrl`, `tenantID`, and corresponding TLS configuration. Cluster roles and a cluster role binding are created for reading and writing logs to Loki. And `authToken` is set to `FORWARD`. You can set these manually using the `Manual` mode.
```
If Kafka is added, will this new mode automatically update the clusterRoleBinding to account for the change from `flowlogs-pipeline` to `flowlogs-pipeline-transformer`?

nathan-weinberg reviewed (commented):
/label qe-approved

skrthomas commented:
One thing I need to add to this PR is making sure CRD values are camelCase, per https://issues.redhat.com/browse/NETOBSERV-1374

jeana-redhat commented on network_observability/network-observability-overview.adoc:
```diff
@@ -37,7 +37,41 @@ In *Observe* -> *Dashboards*, the *Netobserv* dashboard provides a quick overvie
+
+Available metrics names are as follows (names followed by * are enabled by default): 
+
+* namespace_egress_bytes_total
```
Would it make sense to list all these guys in `monospace` font?
```suggestion
* `namespace_egress_bytes_total`
```

skrthomas replied:
I think it would, yea. Bc they will all be specified in a Prometheus definition file. Good catch. Also FYI I am moving this content over to another PR: https://github.com/openshift/openshift-docs/pull/69106/files .

skrthomas replied:
I'll incorporate this feedback over there. TY!

jeana-redhat commented on network_observability/network-observability-overview.adoc:
```diff
@@ -37,7 +37,41 @@ In *Observe* -> *Dashboards*, the *Netobserv* dashboard provides a quick overvie
+* workload_ingress_packets_total
+* workload_flows_total
+
+When the `PacketDrop` feature is enabled in `spec.agent.ebpf.features` (with `privileged` mode), additional metrics are available:
```
```suggestion
When the `PacketDrop` feature is enabled in `spec.agent.ebpf.features` (with `privileged` mode), the following additional metrics are available:
```

skrthomas replied:
I'll incorporate this in this PR: https://github.com/openshift/openshift-docs/pull/69106/files

jeana-redhat commented on network_observability/network-observability-overview.adoc:
```diff
@@ -37,7 +37,41 @@ In *Observe* -> *Dashboards*, the *Netobserv* dashboard provides a quick overvie
+* workload_drop_bytes_total
+* workload_drop_packets_total
+
+When the `FlowRTT` feature is enabled in `spec.agent.ebpf.features`, additional metrics are available:
```
```suggestion
When the `FlowRTT` feature is enabled in `spec.agent.ebpf.features`, the following additional metrics are available:
```

jeana-redhat reviewed (commented):
Just a few things here, overall LGTM.

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

skrthomas commented on modules/network-observability-SRIOV-configuration.adoc:
```diff
@@ -21,7 +21,7 @@ In order to collect traffic from a cluster with a Single Root I/O Virtualization
 .Configure `FlowCollector` for SR-IOV monitoring
```
To be honest, I do not know 🤔 I guess I thought I needed them because of the `.Configure `FlowCollector` for SR-IOV monitoring` YAML heading. I don't think its hurting anything per say, but it may also be pointless.

skrthomas commented on modules/network-observability-disabling-health-alerts.adoc:
```diff
@@ -13,7 +13,7 @@ You can opt out of health alerting by editing the `FlowCollector` resource:
 . Add `spec.processor.metrics.disableAlerts` to disable health alerts, as in the following YAML sample:
 [source,yaml]
```
Good catch :)

skrthomas commented on modules/network-observability-multitenancy.adoc:
Ooops, that's because my metadata is wrong here. Its in `network_observability/installing-operators.adoc` : https://67836--docspreview.netlify.app/openshift-enterprise/latest/network_observability/installing-operators#network-observability-multi-tenancynetwork_observability

skrthomas commented on modules/network-observability-packet-drops.adoc:
```diff
@@ -22,15 +22,15 @@ CPU and memory usage increases when this feature is enabled.
+  deploymentModel: Direct
   agent:
-    type: EBPF
+    type: Ebpf
```
@jpinsonneau I just want to check, is this `Ebpf` or `eBPF`? 👀 https://github.com/netobserv/network-observability-operator/pull/511/files#diff-075e1f2f99419c9e546c4d913bf0ae45a12261f069cb18fcb0046c6184558449R9139

jpinsonneau replied:
That should be `eBPF` indeed ! Mixed capitalization should be preserved for names:
https://github.com/netobserv/network-observability-operator/pull/511/files#diff-6144d5c1ab9988d6897de59fe0ad2b6790dd9d0178b8a3b8bca03c628b4fdcf2

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants

I've updated the sample which was incorrect:
https://issues.redhat.com/browse/NETOBSERV-1374
https://github.com/netobserv/network-observability-operator/pull/518

Thanks for pointing that @skrthomas !

jpinsonneau replied:
```suggestion
    type: eBPF
```

skrthomas replied:
Thanks @jpinsonneau ! I'm glad I can help :)

jpinsonneau commented on modules/network-observability-dns-tracking.adoc:
```diff
@@ -21,15 +21,15 @@ CPU and memory usage increases are observed in the eBPF agent when this feature
+  deploymentModel: Direct
   agent:
-    type: EBPF
+    type: Ebpf
```
```suggestion
    type: eBPF
```

jpinsonneau commented on modules/network-observability-enriched-flows.adoc:
```diff
@@ -20,19 +20,19 @@ You can send network flows to Kafka, IPFIX, or both at the same time. Any proces
         tls:
           enable: false                 <2>
-  - type: IPFIX                         <3>
+  - type: Ipfix                         <3>
```
```suggestion
  - type: IPFIX                         <3>
```

jpinsonneau commented on modules/network-observability-flowcollector-view.adoc:
```diff
@@ -88,5 +79,5 @@ spec:
 <3> The optional specifications `spec.processor.logTypes`, `spec.processor.conversationHeartbeatInterval`, and `spec.processor.conversationEndTimeout` can be set to enable conversation tracking. When enabled, conversation events are queryable in the web console. The values for `spec.processor.logTypes` are as follows: `FLOWS` `CONVERSATIONS`, `ENDED_CONVERSATIONS`, or `ALL`. Storage requirements  ...
 <4> The Loki specification, `spec.loki`, specifies the Loki client. The default values match the Loki install paths mentioned in the Installing the Loki Operator section. If you used another installation method for Loki, specify the appropriate client information for your install.
-<5> The original certificates are copied to the Network Observability instance namespace and watched for updates. When not provided, the namespace defaults to be the same as "spec.namespace". If you chose to install Loki in a different namespace, you must specify it in the `spec.loki.tls.caCert.namespace` field.   Similarly, the `spec.exporters.kafka.tls.caCert.namespace` field is available for K ...
+<5> The `LokiStack` mode automatically sets a few configurations: `querierUrl`, `ingesterUrl` and `statusUrl`, `tenantID`, and corresponding TLS configuration. Cluster roles and a cluster role binding are created for reading and writing logs to Loki. And `authToken` is set to `FORWARD`. You can set these manually using the `Manual` mode.
```
```suggestion
<5> The `LokiStack` mode automatically sets a few configurations: `querierUrl`, `ingesterUrl` and `statusUrl`, `tenantID`, and corresponding TLS configuration. Cluster roles and a cluster role binding are created for reading and writing logs to Loki. And `authToken` is set to `Forward`. You can set these manually using the `Manual` mode.
```

skrthomas replied:
AH I changed this yesterday but I realize I didn't ctrl+s the file in VSCode so it didn't update here 🤦‍♀️

jpinsonneau commented on modules/network-observability-working-with-conversations.adoc:
```diff
@@ -19,22 +19,21 @@ As an administrator, you can you can group network flows that are part of the sa
 ----
 <1> The *Conversation end* event represents the point when the `conversationEndTimeout` is reached or the TCP flag is intercepted.
-<2> When `logTypes` is set to `FLOWS`, only the *Flow* event is exported. If you set the value to `ALL`, both conversation and flow events are exported and visible in the *Network Traffic* page. To focus only on conversation events, you can specify `CONVERSATIONS` which exports the *Conversation start*, *Conversation tick* and *Conversation end* events; or `ENDED_CONVERSATIONS` exports only the * ...
+<2> When `logTypes` is set to `Flows`, only the *Flow* event is exported. If you set the value to `ALL`, both conversation and flow events are exported and visible in the *Network Traffic* page. To focus only on conversation events, you can specify `CONVERSATIONS` which exports the *Conversation start*, *Conversation tick* and *Conversation end* events; or `ENDED_CONVERSATIONS` exports only the * ...
```
```suggestion
<2> When `logTypes` is set to `Flows`, only the *Flow* event is exported. If you set the value to `All`, both conversation and flow events are exported and visible in the *Network Traffic* page. To focus only on conversation events, you can specify `Conversations` which exports the *Conversation start*, *Conversation tick* and *Conversation end* events; or `EndedConversations` exports only the *Conversation end* events. Storage requirements are highest for `All` and lowest for `EndedConversations`.
```

jpinsonneau replied:
Same as Kafka, the notes below should be updated:
```
If you update the `logType` option, the flows from the previous selection do not clear from the console plugin. For example, if you initially set `logType` to `Conversations` for a span of time until 10 AM and then move to `EndedConversations`, the console plugin shows all conversation events before 10 AM and only ended conversations after 10 AM.
```

skrthomas replied:
Thanks for pointing these out!

jpinsonneau commented on modules/network-observability-flowcollector-kafka-config.adoc:
```diff
@@ -20,15 +20,14 @@ You can configure the `FlowCollector` resource to use Kafka for high-throughput
   name: cluster
 spec:
-  deploymentModel: KAFKA                                    <1>
+  deploymentModel: Kafka                                    <1>
```
I can't suggest update from here directly since it's not updated code but you sould also update the <1> mention below:
```
<1> Set `spec.deploymentModel` to `Kafka` instead of `Direct` to enable the Kafka deployment model.
```

skrthomas replied:
Got it, thanks for pointing this one out as well.

jpinsonneau commented on modules/network-observability-SRIOV-configuration.adoc:
```diff
@@ -16,12 +16,11 @@ In order to collect traffic from a cluster with a Single Root I/O Virtualization
 [source,yaml]
 ----
-apiVersion: flows.netobserv.io/v1alpha1
+apiVersion: flows.netobserv.io/v1beta2
```
If we update this we also need to update below: 
```
  deploymentModel: Direct
  agent:
    type: eBPF
```

jpinsonneau commented on modules/network-observability-flowcollector-view.adoc:
```diff
@@ -45,19 +45,10 @@ spec:
       limits:
         memory: 800Mi
     conversationEndTimeout: 10s
-    logTypes: FLOWS                            <3>
+    logTypes: Flows                            <3>
     conversationHeartbeatInterval: 30s
```
I'm still waiting for https://github.com/netobserv/network-observability-operator/pull/467 to be merged here.
`conversationEndTimeout` and `conversationHeartbeatInterval` are part of the fields moving to `debug` sections.

See the [entire list in Jira](https://issues.redhat.com/browse/NETOBSERV-1092)

skrthomas replied:
ACK, will be on the lookout for this to merge and remove these fields. The API doc regeneration PR (yet to be opened) will have the fields.

jpinsonneau replied:
yeah, relying on the API autogenerated doc shoud be good enough here :+1:

jpinsonneau replied:
FYI the `debug` fields has been renamed to `advanced` in the PR above :eyes:

jpinsonneau commented on modules/network-observability-flowcollector-view.adoc:
```diff
@@ -86,7 +77,7 @@ spec:
 <1> The Agent specification, `spec.agent.type`, must be `EBPF`. eBPF is the only {product-title} supported option.
 <2> You can set the Sampling specification, `spec.agent.ebpf.sampling`, to manage resources. Lower sampling values might consume a large amount of computational, memory and storage resources. You can mitigate this by specifying a sampling ratio value. A value of 100 means 1 flow every 100 is sampled. A value of 0 or 1 means all flows are captured. The lower the value, the increase in returned flo ...
-<3> The optional specifications `spec.processor.logTypes`, `spec.processor.conversationHeartbeatInterval`, and `spec.processor.conversationEndTimeout` can be set to enable conversation tracking. When enabled, conversation events are queryable in the web console. The values for `spec.processor.logTypes` are as follows: `FLOWS` `CONVERSATIONS`, `ENDED_CONVERSATIONS`, or `ALL`. Storage requirements  ...
+<3> The optional specifications `spec.processor.logTypes`, `spec.processor.conversationHeartbeatInterval`, and `spec.processor.conversationEndTimeout` can be set to enable conversation tracking. When enabled, conversation events are queryable in the web console. The values for `spec.processor.logTypes` are as follows: `Flows` `Conversations`, `EndedConversations`, or `ALL`. Storage requirements a ...
```
```suggestion
<3> The Processor specification `spec.processor.logTypes` can be set to enable conversation tracking. When enabled, conversation events are queryable in the web console. The values for `spec.processor.logTypes` are as follows: `Flows` `Conversations`, `EndedConversations`, or `ALL`. Storage requirements are highest for `All` and lowest for `EndedConversations`.
```

jpinsonneau commented on modules/network-observability-working-with-conversations.adoc:
```diff
@@ -19,27 +19,26 @@ As an administrator, you can you can group network flows that are part of the sa
 spec:
  processor:
   conversationEndTimeout: 10s                  <1>
-  logTypes: FLOWS                              <2>
+  logTypes: Flows                              <2>
   conversationHeartbeatInterval: 30s           <3>
```
Same as https://github.com/openshift/openshift-docs/pull/67836/files#r1442659471

skrthomas commented:
- [FlowCollector YAML sample](https://github.com/openshift/openshift-docs/pull/67836/files#diff-6a242d769d7901fdf3ab2ccd435524ae6ef219b9b1bb26aacdc2d386bd3b2fbfR47-R52)
- [FlowCollector YAML Code callout](https://github.com/openshift/openshift-docs/pull/67836/files#diff-6a242d769d7901fdf3ab2ccd435524ae6ef219b9b1bb26aacdc2d386bd3b2fbfR81-R82)
- [Working with conversation tracking YAML configuration](https://github.com/openshift/openshift-docs/pull/67836/files#diff-6a242d769d7901fdf3ab2ccd435524ae6ef219b9b1bb26aacdc2d386bd3b2fbfR81-R82)

skrthomas commented on modules/network-observability-working-with-conversations.adoc:
```diff
@@ -19,27 +19,26 @@ As an administrator, you can you can group network flows that are part of the sa
 [NOTE]
 ====
-If you update the `logType` option, the flows from the previous selection do not clear from the console plugin. For example, if you initially set `logType` to `CONVERSATIONS` for a span of time until 10 AM and then move to `ENDED_CONVERSATIONS`, the console plugin shows all conversation events before 10 AM and only ended conversations after 10 AM.
+If you update the `logType` option, the flows from the previous selection do not clear from the console plugin. For example, if you initially set `logType` to `Conversations` for a span of time until 10 AM and then move to `EndedConversations`, the console plugin shows all conversation events before 10 AM and only ended conversations after 10 AM.
```
@jpinsonneau just checking, this note doesn't need to be updated right? Is `EndedCoversations` still a logType: 
![image](https://github.com/openshift/openshift-docs/assets/77501157/c8177831-8dcb-4017-944b-861a8f86eeb1)

jpinsonneau replied:
These didn't changed indeed:
https://github.com/netobserv/network-observability-operator/blob/release-1.5/apis/flowcollector/v1beta1/flowcollector_types.go#L377-L382

```
	// `logTypes` defines the desired record types to generate. Possible values are:<br>
	// - `FLOWS` (default) to export regular network flows<br>
	// - `CONVERSATIONS` to generate events for started conversations, ended conversations as well as periodic "tick" updates<br>
	// - `ENDED_CONVERSATIONS` to generate only ended conversations events<br>
	// - `ALL` to generate both network flows and all conversations events<br>
	// +kubebuilder:validation:Optional
	// +kubebuilder:validation:Enum:="FLOWS";"CONVERSATIONS";"ENDED_CONVERSATIONS";"ALL"
	// +kubebuilder:default:=FLOWS
	LogTypes *string `json:"logTypes,omitempty"`
```

skrthomas replied:
Cool, TY for confirming :)

mburke5678 commented:
@skrthomas Can you squash the commits?

skrthomas commented:
@mburke5678 Yep, sure can. Sorry about that.I didn't realize there were 2 commits here. Must've happened when I rebased and I didn't realize it.
